### PR TITLE
bring back support for disabling progressbar in footer

### PR DESCRIPTION
**Pull request by houqp**
fix https://github.com/koreader/koreader/issues/2361

### CODE REVIEW

File: frontend/apps/reader/modules/readerfooter.lua [hunk 1]
```diff
@@ -255,7 +258,12 @@ function ReaderFooter:resetLayout()
     if new_screen_width == self._saved_screen_width then return end
     local new_screen_height = Screen:getHeight()
 
-    self.progress_bar.width = math.floor(new_screen_width - self.text_width - self.horizontal_margin*2)
+    if self.settings.disable_progress_bar then
+        self.progress_bar.width = 0
+    else
+        self.progress_bar.width = math.floor(
+            new_screen_width - self.text_width - self.horizontal_margin*2)
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerfooter.lua]: Is it on purpose that it's not `new_screen_width - self.text_width - self.horizontal_margin * 2)`? (with spaces around `*`)

File: frontend/apps/reader/modules/readerfooter.lua [hunk 2]
```diff
@@ -500,13 +526,22 @@ end
...
+        self.progress_bar.width = 0
     else
-        self.text_width = self.footer_text:getSize().w + self.text_left_margin
+        if self.has_no_mode then
+            self.text_width = 0
+        else
+            self.text_width = self.footer_text:getSize().w + self.text_left_margin
+        end
+        self.progress_bar.width = math.floor(
+            self._saved_screen_width - self.text_width - self.horizontal_margin*2)
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerfooter.lua]: same as above

File: frontend/apps/reader/modules/readerfooter.lua (hunk 1 above)
Comment (houqp) [inline frontend/apps/reader/modules/readerfooter.lua]: it's on purpose to make it easier to read (at least for me). we can also remove all the spaces if you prefer that.

Comment (Frenzie) [inline frontend/apps/reader/modules/readerfooter.lua]: no, it's fine

**Pull request merged**